kernel/queue.c

```diff
@@ -274,31 +274,39 @@ void k_queue_merge_slist(struct k_queue *queue, sys_slist_t *list)
 static void *k_queue_poll(struct k_queue *queue, s32_t timeout)
 {
 	struct k_poll_event event;
-	int err;
+	int err, elapsed = 0, done = 0;
```

Comment:
done should be bool? Or is MISRA ~stupidity~ smartness?

Reply:
> Or is MISRA ~stupidity~ smartness?

https://stackoverflow.com/a/30614236

Reply:
It's just a style choice on my part.  C in fact provides no particular typesafety here, the returned type of comparison operators is int.  Machine words as default booleans have a long tradition and are used pervasively in the kernel.  I'm certainly not chasing a MISRA standard, I promise.